contracts/FlyoverDiscovery.sol

```diff
@@ -0,0 +1,266 @@
+        for (uint i = 1; i <= lastProviderId; ++i) {
+            if (_shouldBeListed(_liquidityProviders[i])) {
+                ++count;
+            }
+        }
+        Flyover.LiquidityProvider[] memory providersToReturn = new Flyover.LiquidityProvider[](count);
+        count = 0;
+        for (uint i = 1; i <= lastProviderId; ++i) {
+            lp = _liquidityProviders[i];
+            if (_shouldBeListed(lp)) {
+                providersToReturn[count] = lp;
+                ++count;
+            }
+        }
+        return providersToReturn;
+    }
+
+    /// @notice Returns a single LP by address
+    /// @param providerAddress The LP address
+    /// @return provider LP record, reverts if not found
+    function getProvider(address providerAddress) external view returns (Flyover.LiquidityProvider memory) {
+        return _getProvider(providerAddress);
+    }
+
+    /// @notice Checks if an LP can operate for peg-in side
+    /// @dev Ignores the first argument as compatibility stub with legacy signature
+    /// @param addr The LP address
+    /// @return isOp True if registered and peg-in collateral >= min
+    function isOperational(Flyover.ProviderType, address addr) external view returns (bool) {
+        return _isRegistered(addr) && _collateralManagement.getPegInCollateral(addr) >= _minCollateral;
+    }
+
+    /// @notice Checks if an LP can operate for peg-out side
+    /// @param addr The LP address
+    /// @return isOp True if registered-for-pegout and peg-out collateral >= min
+    function isOperationalForPegout(address addr) external view returns (bool) {
+        return _isRegisteredForPegout(addr) && _collateralManagement.getPegOutCollateral(addr) >= _minCollateral;
+    }
+
+    // ------------------------------------------------------------
+    // Getter Functions
+    // ------------------------------------------------------------
+
+    /// @notice Returns the current minimum collateral threshold per side
+    /// @return minCollateral The minimum collateral value
+    function getMinCollateral() external view returns (uint256) {
+        return _minCollateral;
+    }
+
+    /// @notice Returns the last assigned provider id
+    /// @return lastId Last provider id
+    function getProvidersId() external view returns (uint) {
+        return lastProviderId;
+    }
+
+    /// @notice Returns the resignation starting block for an LP
+    /// @param addr The LP address
+    /// @return resignationBlock Block number when resignation started, or 0 if not resigned
+    function getResignationBlockNum(address addr) external view returns (uint256) {
+        return _resignationBlockNum[addr];
+    }
+
+    // ------------------------------------------------------------
+    // FlyoverDiscovery Private Functions
+    // ------------------------------------------------------------
+
+    function _shouldBeListed(Flyover.LiquidityProvider storage lp) private view returns(bool){
```

Comment:
same here we can use the PoC version https://github.com/rsksmart/liquidity-bridge-contract/blob/lbc-split/contracts/split/FlyoverDiscovery.sol#L116